app/scripts/lib/tx-state-manager.js

```diff
@@ -0,0 +1,230 @@
+    const txMeta = this.getTx(txId)
+    txMeta.txParams = extend(txMeta.txParams, txParams)
+    this.updateTx(txMeta)
+  }
+
+/*
+  Takes an object of fields to search for eg:
+  let thingsToLookFor = {
+    to: '0x0..',
+    from: '0x0..',
+    status: 'signed',
+    err: undefined,
+  }
+  and returns a list of tx with all
+  options matching
+
+  ****************HINT****************
+  | `err: undefined` is like looking |
+  | for a tx with no err             |
+  | so you can also search txs that  |
+  | dont have something as well by   |
+  | setting the value as undefined   |
+  ************************************
+
+  this is for things like filtering a the tx list
+  for only tx's from 1 account
+  or for filltering for all txs from one account
+  and that have been 'confirmed'
+  */
+  getFilteredTxList (opts, initialList) {
+    let filteredTxList = initialList
+    Object.keys(opts).forEach((key) => {
+      filteredTxList = this.getTxsByMetaData(key, opts[key], filteredTxList)
+    })
+    return filteredTxList
+  }
+
+  getTxsByMetaData (key, value, txList = this.getTxList()) {
+    return txList.filter((txMeta) => {
+      if (txMeta.txParams[key]) {
+        return txMeta.txParams[key] === value
+      } else {
+        return txMeta[key] === value
+      }
+    })
+  }
+
+  // STATUS METHODS
+  // statuses:
+  //    - `'unapproved'` the user has not responded
+  //    - `'rejected'` the user has responded no!
+  //    - `'approved'` the user has approved the tx
+  //    - `'signed'` the tx is signed
+  //    - `'submitted'` the tx is sent to a server
+  //    - `'confirmed'` the tx has been included in a block.
+  //    - `'failed'` the tx failed for some reason, included on tx data.
+
+  // get::set status
+
+  // should return the status of the tx.
+  getTxStatus (txId) {
+    const txMeta = this.getTx(txId)
+    return txMeta.status
+  }
+
+  // should update the status of the tx to 'rejected'.
+  setTxStatusRejected (txId) {
+    this._setTxStatus(txId, 'rejected')
+  }
+
+  // should update the status of the tx to 'approved'.
+  setTxStatusApproved (txId) {
+    this._setTxStatus(txId, 'approved')
+  }
+
+  // should update the status of the tx to 'signed'.
+  setTxStatusSigned (txId) {
+    this._setTxStatus(txId, 'signed')
+  }
+
+  // should update the status of the tx to 'submitted'.
+  setTxStatusSubmitted (txId) {
+    this._setTxStatus(txId, 'submitted')
+  }
+
+  // should update the status of the tx to 'confirmed'.
+  setTxStatusConfirmed (txId) {
+    this._setTxStatus(txId, 'confirmed')
+  }
+
+  setTxStatusFailed (txId, err) {
+    const txMeta = this.getTx(txId)
+    txMeta.err = {
+      message: err.toString(),
+      stack: err.stack,
+    }
+    this.updateTx(txMeta)
+    this._setTxStatus(txId, 'failed')
+  }
+
+/* _____________________________________
+|                                      |
+|           PRIVATE METHODS            |
+|______________________________________*/
```

Comment:
our code style for comment sections in other places has been
```
//
/// title
//
```

Reply:
so this is from a while ago i made this big and bold because their were some non private looking functions in the private section so this was a little louder but if you are so against it ill change it

Reply:
✔️